test/unit/math/mix/fun/complex_schur_decompose_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include <test/unit/math/test_ad.hpp>
+#include <stdexcept>
+
+TEST(mathMixFun, complexSchurDecomposeT) {
+  auto f = [](const auto& x) {
+    using stan::math::complex_schur_decompose_t;
+    return complex_schur_decompose_t(x);
+  };
+  for (const auto& x : stan::test::square_test_matrices(0, 2)) {
+    stan::test::expect_ad(f, x);
+  }
+
+  Eigen::MatrixXd a32(3, 2);
+  a32 << 3, -5, 7, -7.2, 9.1, -6.3;
+  EXPECT_THROW(f(a32), std::invalid_argument);
+}
+
+TEST(mathMixFun, complexSchurDecomposeU) {
+  auto f = [](const auto& x) {
+    using stan::math::complex_schur_decompose_u;
+    return complex_schur_decompose_u(x);
+  };
+  for (const auto& x : stan::test::square_test_matrices(0, 2)) {
+    stan::test::expect_ad(f, x);
+  }
+
+  Eigen::MatrixXd a32(3, 2);
+  a32 << 3, -5, 7, -7.2, 9.1, -6.3;
+  EXPECT_THROW(f(a32), std::invalid_argument);
+}
+
+template <typename V>
+void test_complex_schur_decompose(const Eigen::MatrixXd& x) {
+  using stan::math::complex_schur_decompose_t;
+  using stan::math::complex_schur_decompose_u;
+  using stan::math::value_of_rec;
+  Eigen::Matrix<V, -1, -1> X(x.rows(), x.cols());
+  for (int i = 0; i < x.size(); ++i)
```

Comment:
Why isn't this just `X = x`?

Reply:
My guess was that because `x` is a Matrix of double and X could be a matrix of autodiff variables that we'd need elementwise assignment to handle that promotion, but it seems that isn't true (it compiles if I change it, anyway)